File: CS50/week6/pset6/cash.py
from cs50 import get_float

def main():
    while True:
        amount = get_float("Change owed: ")
        if (amount > 0):
            break

    counter = 0

    for x in [0.25, 0.10, 0.05, 0.01]:
        while amount >= x:
            amount = round(amount - x, 2)
            counter += 1

    print(f"The minimum amount of coins is {counter}")

if __name__ == "__main__":
    main()

File: CS50/week1/pset1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float money;

    // Validate user input
    do
    {
        money = get_float("Change owed: ");
    }
    while (money < 0);

    int intMoney = round(money * 100);

    int counter = 0;

    while (intMoney >= 25)
    {
        intMoney = intMoney - 25;
        counter++;
    }
    while (intMoney >= 10)
    {
        intMoney = intMoney - 10;
        counter++;
    }
    while (intMoney >= 5)
    {
        intMoney = intMoney - 5;
        counter++;
    }
    while (intMoney >= 1)
    {
        intMoney = intMoney - 1;
        counter++;
    }
    printf("%i\n", counter);
}
